kernel/kernel-impl/src/main/java/org/sakaiproject/messaging/impl/UserMessagingServiceImpl.java

```diff
@@ -69,9 +82,35 @@ public class UserMessagingServiceImpl implements UserMessagingService {
     @Setter    private ResourceLoader resourceLoader;
 
     private ExecutorService executor;
+    private PushService pushService;
+    private String publicKey = "";
+    private boolean pushEnabled = false;
+
+    private final static ObjectMapper objectMapper = new ObjectMapper();
 
     public void init() {
+
+        Security.addProvider(new BouncyCastleProvider());
+
         executor = Executors.newFixedThreadPool(20);
+
+        this.pushEnabled = serverConfigurationService.getBoolean("portal.notifications.push.enabled", false);
+
+        if (pushEnabled) {
+            String home = serverConfigurationService.getSakaiHomePath();
+
+            String publicKeyFileName = serverConfigurationService.getString("portal.notifications.push.publickey", "sakai_push.key.pub");
+            String privateKeyFileName = serverConfigurationService.getString("portal.notifications.push.privatekey", "sakai_push.key");
+            try {
+                publicKey = String.join("", Files.readAllLines(Paths.get(home, publicKeyFileName)));
+                String privateKey = String.join("", Files.readAllLines(Paths.get(home, privateKeyFileName)));
+                pushService = new PushService(publicKey, privateKey);
+                String pushSubject = serverConfigurationService.getString("portal.notifications.push.subject", "");
```

Comment:
I'd think some default subject should be used maybe like "Sakai Push Notification"? Maybe it could be pulled out of i18n? How does this subject message look on the user side?

Reply:
User doesn't see it. The service worker processes the push messages from the push server and just pulls the data it wants. I'll update that subject to be that default, that's a good point, thanks.